infra/helper.py

```diff
@@ -616,26 +613,13 @@ def build_image_impl(project, cache=True, pull=False, architecture='x86_64'):
 
   build_args = []
   image_name = 'gcr.io/%s/%s' % (image_project, image_name)
-  if architecture == 'aarch64':
```

Comment:
Why did you need to remove this? I think doing so breaks aarch64 builds on x86_64 hosts.

Reply:
when i use build_fuzzer command as below:
```
python3.10 infra/helper.py build_fuzzers --architecture aarch64 --sanitizer address libtiff
```
I get following error on Mac M2:
```
unknown flag: --platform
See 'docker --help'.
```

looks like some issue with docker command, here is the detailed log:
```
Run 'docker COMMAND --help' for more information on a command.

To get more help with docker, check out our guides at https://docs.docker.com/go/guides/

Traceback (most recent call last):
  File "/home/parallels/Research/oss-fuzz/infra/helper.py", line 1674, in <module>
    sys.exit(main())
  File "/home/parallels/Research/oss-fuzz/infra/helper.py", line 185, in main
    result = build_fuzzers(args)
  File "/home/parallels/Research/oss-fuzz/infra/helper.py", line 926, in build_fuzzers
    return all(
  File "/home/parallels/Research/oss-fuzz/infra/helper.py", line 927, in <genexpr>
    build_fuzzers_impl(args.project,
  File "/home/parallels/Research/oss-fuzz/infra/helper.py", line 775, in build_fuzzers_impl
    if build_project_image and not build_image_impl(project,
  File "/home/parallels/Research/oss-fuzz/infra/helper.py", line 637, in build_image_impl
    subprocess.check_call(command)
  File "/usr/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['docker', 'buildx', 'build', '--platform', 'linux/arm64', '--progress', 'plain', '--load', '-t', 'gcr.io/oss-fuzz/libtiff', '--file', '/home/parallels/Research/oss-fuzz/projects/libtiff/Dockerfile', '/home/parallels/Research/oss-fuzz/projects/libtiff']' returned non-zero exit status 125.
```
when i remove, its working fine and i can build image/fuzzers and can run them.

As this change is breaking aarch64 build on x86_64 hosts, we can specify a arg "--cross_compile=1" which can be used in case we want to generate aarch64 build on x86_64. If we just want to create aarch64 build on arch64 itself we dont need to use "--cross_compile" args at all? does it make sense?

Reply:
> unknown flag: --platform

I think it indicates that it isn't supported by docker. Assuming this is run in the guest VM my guess would be that either the guest OS is out of date there or docker was installed from out-of-date repositories somehow (or both).

That being said I think the right fix would be to not pass `--platform` to `docker` unless `--architecture` is passed to `helper.py` explicitly. This way `helper.py` would always build fuzz targets using the native architecture by default.

Reply:
> Assuming this is run in the guest VM

FWIW if it's run on the host it could be that docker was installed via `brew install docker` and that version of `docker` doesn't support the platform option. `brew install --cask docker` should install the official docker app where the platform option should work. Either way that error message indicates that `docker` isn't compatible with `helper.py` in its current form.

Reply:
It was run on guest Linux vm. It's Kali Linux and docker was installed using apt install. 
Yes helper.py is not compatible with the docker it seems.
Will check the docker version and provide here so that it can help.

Reply:
As far as I can remember there are two ways to install Docker on Debian-based distributions. I think `docker-ce` should work
https://docs.docker.com/engine/install/debian/#uninstall-old-versions
https://www.kali.org/docs/containers/installing-docker-on-kali/#installing-docker-ce-on-kali-linux.